### ISSUE TITLE
Closing magics existing connection

### ISSUE THREAD

**Issue by idomic** [ploomber/jupysql#37]
Essentially there's no way of closing a magics initiated connection.
We should fix this [issue thread sample](https://github.com/catherinedevlin/ipython-sql/issues/65).

**Comment by idomic** [comment]
@meihkv feel free to submit a PR we're working on updating the package and maintaining it via a fork.
cc: @ecrooks @misurin @pkasinathan @psychemedia @das-g @fabianhauser @memeplex @ParfaitG @ken-approveshield

**Comment by meihkv** [comment]
Sure thing!

**Comment by das-g** [comment]
Why was I CC-mentioned here? I feel like I'm missing some context.

**Comment by das-g** [comment]
> Why was I CC-mentioned here? I feel like I'm missing some context.

Ah, probably because I've upvoted catherinedevlin/ipython-sql#65.

**Comment by idomic** [comment]
@das-g exactly :)
@meihkv awesome! Please tag when ready for review.

**Comment by edublancas** [comment]
This has already been fixed in ipython-sql via the `-x`/`--close` option. However, Iooked at the code and the only way to close a connection is to pass the connection string, which is odd; there's room for improvement here.

One thing that comes to mind is to allow identifying connections with keys:

```python
%sql {some-connection-string} --name some-name
%sql {another-connection-string} --name another-name
```

And then allowing them to be closed with the name:

```python
%sql --close some-name
%sql --close another-name
```

This would also allow users to switch connections more naturally:

```python
%%sql some-name
SELECT * FROM some_table
```

```python
%%sql another-name
SELECT * FROM another_table
```

**Comment by idomic** [comment]
I like the names idea, I think if a user don't give it a name, we should generate a random one for the user and print it instead of the connection string. Probably want to have `-n` as well.

**Comment by edublancas** [comment]
Documentation:

- https://jupysql.readthedocs.io/en/latest/api.html#connect-to-database
- https://jupysql.readthedocs.io/en/latest/api.html#close-connection

**Comment by idomic** [comment]
@meihkv @ecrooks @misurin @pkasinathan @psychemedia @das-g @fabianhauser @memeplex @ParfaitG @ken-approveshield
We just updated the docs and released a new version( [0.5.2 changelog](https://github.com/ploomber/jupysql/blob/0.5.2/CHANGELOG.md)).
Make sure to [join our slack](https://ploomber.io/community/) for more info and support
